packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -215,9 +215,10 @@ export type ObjectPlugin<Options = any[]> = {
 export type FunctionPlugin<Options = any[]> = PluginInstallFunction<Options> &
   Partial<ObjectPlugin<Options>>
 
-export type Plugin<Options = any[]> =
-  | FunctionPlugin<Options>
-  | ObjectPlugin<Options>
+export type Plugin<
+  Options = any[],
```

Comment:
What do you think about a note here to change options so in the next major the P can be removed?

```suggestion
  // TODO: in next major Options extends unknown[] and remove P
  Options = any[],
```